src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -31,16 +32,17 @@ class StartableEndpoint : IStartableEndpoint
             this.builder = builder;
         }
 
-        public async Task<IEndpointInstance> Start()
+        public async Task<IEndpointInstance> Start(CancellationToken cancellationToken = default)
         {
-            var transportInfrastructure = await transportSeam.CreateTransportInfrastructure().ConfigureAwait(false);
+            var transportInfrastructure = await transportSeam.CreateTransportInfrastructure(cancellationToken).ConfigureAwait(false);
 
             var pipelineCache = pipelineComponent.BuildPipelineCache(builder);
             var messageOperations = sendComponent.CreateMessageOperations(builder, pipelineComponent);
-            var rootContext = new RootContext(builder, messageOperations, pipelineCache);
+            // This context is persisted in the IMessageSession for its lifetime, and should not carry the CancellationToken from Endpoint.Start()
```

Comment:
Correct, if/when we introduce a `IEndpointInstance.OpenSession()` I guess we would accept a token parameter there